Stability: Improve Developer Documentation (#8517)
solo-io/gloo: devel/testing/run-tests.md

# Run Tests
- [Background](#background)
- [Common Make Targets](#common-make-targets)
  - [test](#test)
  - [test-with-coverage](#test-with-coverage)
  - [run-tests](#run-tests)
  - [run-e2e-tests](#run-e2e-tests)
  - [run-kube-e2e-tests](#run-kube-e2e-tests)
  - [run-benchmark-tests](#run-benchmark-tests)
- [Environment Variables](#environment-variables)
  - [GINKGO_USER_FLAGS](#ginkgo_user_flags)
  - [TEST_PKG](#test_pkg)
  - [WAIT_ON_FAIL](#wait_on_fail)
  - [INVALID_TEST_REQS](#invalid_test_reqs)

## Background

> Gloo Edge relies on the [Ginkgo](https://onsi.github.io/ginkgo/) test framework. As outlined in the documentation, it pairs with the [Gomega](https://onsi.github.io/gomega/) matcher library to provide a BDD-style testing framework. For more details about how to write tests, check out the [writing tests docs](writing-tests.md).

Reviewer: ```suggestion
Gloo Edge testing leverages the [Ginkgo](https://onsi.github.io/ginkgo/) test framework. As outlined in the linked documentation, Ginkgo pairs with the [Gomega](https://onsi.github.io/gomega/) matcher library to provide a BDD-style testing framework. For more details about how to write tests, check out our [writing tests docs](writing-tests.md).
```

## Common Make Targets
There are a few common make targets that can be used to run tests

### test
The `test` target provides a wrapper around invoking `ginkgo` with a set of useful flags. This is the base target that is used by all other test targets.

### test-with-coverage
Run tests with coverage reporting

### run-tests

> Run unit tests (non e2e)

Reviewer: ```suggestion
Run unit tests (ie non e2e)
```
